pkg/doc/verifiable/credential.go

```diff
@@ -320,7 +339,7 @@ func defaultCredentialOpts() *credentialOpts {
 		schemaDownloadClient: &http.Client{},
 		disabledCustomSchema: false,
 		decoders:             []CredentialDecoder{decodeIssuer},
-		template:             func() *Credential { return &Credential{} },
+		template:             func(credentialTypes []string) (*Credential, error) { return &Credential{}, nil },
```

Comment:
should the second return type be a boolean or an error?

Reply:
I think that returning an error is more explanative than just boolean value. We can say why the template can't be applied.

Reply:
Well, after consideration I decided to make it internal. No need for making it public really.